tests/mocks.js

```diff
@@ -0,0 +1,349 @@
+    },
+    trace: function _trace() {
+    },
+    debug: function _debug() {
+    },
+    info: function _info() {
+    },
+    warn: function _warn() {
+    },
+    error: function _error(err) {
+        var stderrLines = err.stderrLines;
+
+        if (stderrLines) {
+            if (stderrLines[stderrLines.length - 1]
+                .match(/^Requested unique lookup but found 0 results./)) {
+                // VM doesn't exist, not really an error
+                return;
+            } else if (stderrLines[stderrLines.length - 1]
+                .match(/^ENOENT, open.*\.xml/)) {
+                // VM doesn't exist, not really an error
+                return;
+            }
+        } else if (err.stderr && err.stderr.match(/^ENOENT, open.*\.xml/)) {
+            // VM doesn't exist, not really an error
+            return;
+        }
+        console.log(err); // eslint-disable-line
+    }
+};
+var vmadmVms = [];
+var vmadmErr = null;
+var vmapiVms = [];
+var vmapiGetErr = null;
+var vmapiPutErr = null;
+var vmapiErrVms = {};
+
+
+/*
+ * VMAPI translates the VM objects when reading from Moray (see sdc-vmapi
+ * lib/common/vm-common.js) and we want to be able to make VMs look like they
+ * would from VMAPI so we've copied that logic here.
+ *
+ */
+function vmapifyVm(vmobj) {
+    var defaultFields = JSON.parse(JSON.stringify(VMAPI.VMAPI_DEFAULT_FIELDS));
+    var newObj = JSON.parse(JSON.stringify(vmobj));
+
+    if (newObj.brand === 'kvm') {
+        [
+            'cpu_type',
+            'disks',
+            'vcpus'
+        ].forEach(function _addKvmDefaultField(f) {
+            defaultFields[f] = null;
+        });
+    } else {
+        defaultFields.image_uuid = null;
+    }
+
+    Object.keys(defaultFields).forEach(function _addMissingField(field) {
+        if (!newObj.hasOwnProperty(field)) {
+            newObj[field] = defaultFields[field];
+        }
+    });
+
+    if (!newObj.ram && newObj.max_physical_memory) {
+        newObj.ram = newObj.max_physical_memory;
+    }
+
+    return (newObj);
+}
+
+function vmadmifyVm(vmobj) {
+    var newObj = JSON.parse(JSON.stringify(vmobj));
+
+    assert.uuid(vmobj.uuid, 'vmobj.uuid');
+    assert.string(vmobj.brand, 'vmobj.brand');
+    assert.string(vmobj.state, 'vmobj.state');
+    assert.string(vmobj.zone_state, 'vmobj.zone_state');
+
+    if (!vmobj.snapshots) {
+        newObj.snapshots = [];
+    }
+
+    return (newObj);
+}
+
+
+/*
+ * This coordinator is an event emitter that we use from within the mocks to
+ * tell us when those functions have occurred. Tests can watch for events which
+ * indicate the calling of each function. and the event will include the
+ * relevant function parameters.
+ */
+function Coordinator() {
+    // Initialize necessary properties from `EventEmitter` in this instance
+    EventEmitter.call(this);
+}
+util.inherits(Coordinator, EventEmitter);
+coordinator = new Coordinator();
+
+
+/*
+ * vmadm mock
+ */
+
+function fakeVmadm() {
```

Comment:
Is a constructor really needed? None of the functions set on the famkeVmadm constructor use `this`, so maybe using just plain functions would be clearer? Same question for `fakeVmapi`.

If the goal is to separate these two groups of functions, maybe they should be separate modules instead?

Otherwise, it seems it would be more consistent to set the functions on `fakeVmadm`'s prototype, as it's done for `fakeVmapi`.